src/modules/facility/facility.controller.ts

```diff
@@ -65,4 +76,28 @@ export class FacilityController {
       probabilityOfDefault: facility.probabilityOfDefault,
     };
   }
+
+  @Post()
+  @ApiOperation({ summary: 'Create a new facility.' })
+  @ApiBody({
+    type: CreateFacilityRequestItem,
+    isArray: true,
+  })
+  @ApiCreatedResponse({
+    description: 'The facility has been successfully created.',
+    type: CreateFacilityResponse,
+  })
+  @ApiBadRequestResponse({
+    description: 'Bad request.',
+  })
+  @ApiInternalServerErrorResponse({
+    description: 'An internal server error has occurred.',
+  })
+  async createFacility(
+    @Body(new ParseArrayPipe({ items: CreateFacilityRequestItem })) createFacilityDto: CreateFacilityRequest,
+  ): Promise<CreateFacilityResponse> {
+    const facilityToCreate = createFacilityDto[0];
```

Comment:
looking at the value that we use below, I wonder if we can restructure this into:
```js
const { facilityIdentifier } = createFacilityDto[0];
```
and we replace the return below into
```js
return { facilityIdentifier };
```

Reply:
I also need the full `facilityToCreate` object to pass into `this.facilityService.createFacility`.

I could write it as 
```javascript
const facilityToCreate = createFacilityDto[0];
await this.facilityService.createFacility(facilityToCreate);

const { facilityIdentifier } = facilityToCreate;
return { facilityIdentifier };
```
if you prefer that?

Reply:
ah, you are right! nope, let's leave it as it is